semana7/labefy/src/components/Create.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react'
+import styled from 'styled-components'
+import axios from 'axios'
+
+const MainDiv = styled.div `
+    background-color: #282c34;
+    color: white;
+    margin: 0;
+    padding: 0;
+`
+
+const PlaylistInput = styled.input `
+    margin: 30px;
+`
+const ColoredButton = styled.button `
+background-color: green;
+color: white;
+`
+const axiosConfig = {
+    headers: {
+      Authorization: "rafael-fontes-dumont"
+    }
+}
+
+class Create extends React.Component {
+    state = {
+        PlaylistInputValue: "",
+        
+      }
+
+      onChangePlaylistInput = (event) => {
+        this.setState({PlaylistInputValue: event.target.value})
+    }
+      
+    onClickCreatePlaylist = () => {
+        axios.post("https://us-central1-labenu-apis.cloudfunctions.net/labefy/playlists", 
+        {name: this.state.PlaylistInputValue}, axiosConfig)
+        .then(response => {
+        alert("Lista criada com sucesso!")
+        console.log("lista criada")
+        })
+        .catch(error => {
+        alert("Lista já existente")
+        console.log(error)
+        })
+        this.setState({PlaylistInputValue:""})

```

Comment:
Poderia deletar o `console.log` na linha 40.
Outra coisa que reparei ao criar uma playlist é que o seu input está limpando logo que clico no botão, pois você setou o state na linha 46, fora da função assíncrona.
Acho que seria bacana limpar o input só quando a requisição for concluída com sucesso.
Nesse caso você poderia chamar o `this.setState({PlaylistInputValue:""})` lá em cima, dentro do then, na linha 40.
Dessa forma ele só limpa o input se der tudo certo com a criação da playlist. Se cair em algum erro, seria legal para a(o) usuária(o) que o valor do input ficasse salvo, pra ele poder tentar de novo.
Poderia adicionar uma verificação para testar se o input não está vazio antes de criar a playlist, pois ao clicar no botão sem preencher o nome da playlist ele já está chamando o erro "Lista já existente".
Você poderia colocar um `alert` ou criar uma função separada pra verificar se foi preenchido esse campo antes de fazer a requisição.
Indentar linhas 39 e 40, e linhas 43 e 44.

Reply:
Não tinha reparado nessa questão de limpar o `input`, bem observado.
Tentei criar uma função para checar o `input `vazio, mas não deu certo. Devia ter escolhido outra frase para o `alert ` para não criar confusão